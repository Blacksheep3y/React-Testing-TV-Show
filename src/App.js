// Dependencies (already Imported)
import React, { useState, useEffect } from "react";
import Dropdown from "react-dropdown";
import parse from "html-react-parser";

// File Imports
import { fetchShow } from "./api/fetchShow";

import { formatSeasons } from "./utils/formatSeasons";

import Episodes from "./components/Episodes";
import "./styles.css";

// Default Export
export default function App() {
  const [show, setShow] = useState(null);
  const [seasons, setSeasons] = useState([]);
  const [selectedSeason, setSelectedSeason] = useState("");
  const episodes = seasons[selectedSeason] || [];

  // --- *** Moving the async call *** --- \\

  // useEffect(() => {
  //   const fetchShow = () => {
  //     axios
  //       .get(
  //         "https://api.tvmaze.com/singlesearch/shows?q=stranger-things&embed=episodes"
  //       )
  //       .then(res => {
  //         setShow(res.data);
  //         setSeasons(formatSeasons(res.data._embedded.episodes));
  //       });
  //   };
  //   fetchShow();
  // }, []);

  // *** Recreate the UseEffect
  useEffect(() => {
    fetchShow()
    .then((res) => {
      console.log("res: ", res);
      setShow(res.data);
      setSeasons(formatSeasons(res.data._embedded.episodes));
    })
    .catch((err) => {
      console.log("err ", err);
    });
}, []);

  const handleSelect = e => {
    setSelectedSeason(e.value);
  };

  if (!show) {
    return <h2>Fetching data...</h2>;
  }

  return (
    <div className="App">
      <img className="poster-img" src={show.image.original} alt={show.name} />
      <h1>{show.name}</h1>
      {parse(show.summary)}
      <Dropdown
        options={Object.keys(seasons)}
        onChange={handleSelect}
        value={selectedSeason || "Select a season"}
        placeholder="Select an option"
      />
      <Episodes episodes={episodes} />
    </div>
  );
}
