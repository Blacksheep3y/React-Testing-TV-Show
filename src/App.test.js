// import dependencies
import React from 'react';
import renderer from 'react-test-renderer';

// import files
import App from './App';

//SNAPSHOT TEST
describe('<App />', () => {
    it('App.js matches snapshot', () => {
        const tree = renderer.create(<App />);
        
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
