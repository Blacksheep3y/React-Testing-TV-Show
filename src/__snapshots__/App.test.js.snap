// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App.js matches snapshot 1`] = `
<h2>
  Fetching data...
</h2>
`;
